### DISCUSSION TITLE
Question on rules

### DISCUSSION THREAD

**Discussion by octave2000** [pocketbase/pocketbase#2662] [Q&A]
Is there a way I can formulate a rule that counts number of records in a collection

**Comment by ganigeorgiev** [comment]
There is no direct rule for counting records as it is not clear what would be the criteria for the count (eg. users may want "group by", exclusion criteria, etc.).

The general workaround is to try to create a helper [View collection](https://pocketbase.io/docs/collections/#view-collection) with a count query and then join it with the `@collection.yourViewCollection.*` rule based on your criteria.

For the simplest case when there is no grouping, this could look something like:

1. Create an example `postsCount` View collection:
    ```sql
    SELECT 1 as id, count(*) as total
    FROM posts
    ```

2. Join it with `@collection.postsCount.total` in the collection(s) where you want to check the count:
    ```js
    // posts List/Search API rule
    @collection.postsCount.total > 10 && ...
    ```

---

If your use case require grouping then it is slightly more complicated as it depends on what actually are the criteria. For example if in the above imaginary `posts` collection there was a "category" `relation` field and we want to count only the posts within the same category, the above example could look something like:

1. Create an example `postsCount` View collection:
    ```sql
    -- id        - total
    -- category1 - 10
    -- category2 - 20
    -- etc.
    SELECT
        posts.category as id,
        count(*) as total
    FROM posts
    GROUP BY posts.category
    ```

2. Join it with `@collection.postsCount.*` in the collection(s) where you want to check the count for each matching category:
    ```js
    // posts List/Search API rule
    @collection.postsCount.id ?= category && @collection.postsCount.total ?> 10 && ...
    ```
